### PR TITLE
feat: revoke sessions

### PR DESCRIPTION
**Pull request by tomiir**
# Description

- Adds `wallet_revokePermissions` integration into AppKit.
- Adds corresponding handlers for `wallet_revokePermissions` on clients `ConnectionControllerClient`
- Adds corresponding internal handling in `SessionsController`

- WIP: session revoked successfully screen

## Known Issue
- Cosigner service accepts requests even after revoking. @geekbrother is taking a look

## Type of change

- [ ] Chore (non-breaking change that addresses non-functional tasks, maintenance, or code quality improvements)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Associated Issues

Closes APKT-1341
Closes APKT-1343

# Checklist

- [ ] Code in this PR is covered by automated tests (Unit tests, E2E tests)
- [X] My changes generate no new warnings
- [X] I have reviewed my own code
- [X] I have filled out all required sections
- [ ] I have tested my changes on the preview link
- [ ] Approver of this PR confirms that the changes are tested on the preview link

### CODE REVIEW

File: apps/laboratory/src/utils/ConstantsUtil.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ if (!projectId) {
   throw new Error('NEXT_PUBLIC_PROJECT_ID is not set')
 }
 export const WALLET_URL = process.env['WALLET_URL'] || 'https://react-wallet.walletconnect.com/'
-export const USEROP_BUILDER_SERVICE_BASE_URL = 'https://rpc.walletconnect.com/v1/wallet'
+export const USEROP_BUILDER_SERVICE_BASE_URL = 'https://rpc.walletconnect.org/v1/wallet'
```
Comment (magiziz) [inline apps/laboratory/src/utils/ConstantsUtil.ts:26]: Do we need .org ?

File: packages/appkit/src/universal-adapter/client.ts [hunk 2]
```diff
@@ -289,6 +290,14 @@ export class UniversalAdapterClient {
 
         return provider.request({ method: 'wallet_grantPermissions', params })
       },
+      revokePermissions: async (session: object | readonly unknown[]) => {
+        const provider = await this.getWalletConnectProvider()
+        if (!provider) {
+          throw new Error('connectionControllerClient:grantPermissions - provider is undefined')
+        }
+
+        return provider.request({ method: 'wallet_revokePermissions', params: session })
```
[the remark this answers is not in the document]
Comment (magiziz) [reply packages/appkit/src/universal-adapter/client.ts:299]: I see that session parameter on `revokePermissions` function has `object | readonly unknown[]` type

File: packages/experimental/src/smart-session/ui/views/w3m-smart-session-list-view/index.ts [hunk 3]
```diff
@@ -218,11 +218,22 @@ export class W3mSmartSessionListView extends LitElement {
...
+                  ></wui-permission-contract-call>
+                  ${this.currentTab === 0
+                    ? html`<wui-button
+                        @click=${this.revokePermission.bind(this, session)}
+                        fullWidth
+                        variant="accent"
+                      >
+                        Revoke
+                      </wui-button>`
+                    : ''}
```
Comment (magiziz) [inline packages/experimental/src/smart-session/ui/views/w3m-smart-session-list-view/index.ts:235]: Can we return `null` ?

**Review by magiziz — APPROVED** [review]

**Review by lukaisailovic — APPROVED** [review]

File: apps/laboratory/src/utils/ConstantsUtil.ts (hunk 1 above)
Comment (enesozturk) [reply apps/laboratory/src/utils/ConstantsUtil.ts:26]: @magiziz `org` is including the countries which have blocked `com` domains like China

File: services/id-allocation-service/package.json [hunk 4]
```diff
@@ -9,7 +9,7 @@
     "hono": "4.6.5"
   },
   "devDependencies": {
-    "@cloudflare/workers-types": "4.20240529.0",
+    "@cloudflare/workers-types": "4.20241011.0",
```
Comment (enesozturk) [inline services/id-allocation-service/package.json:12]: Date in version number, interesting approach xd

**Review by enesozturk — APPROVED** [review]

**Pull request merged**